conbench/entities/benchmark_result.py

```diff
@@ -521,6 +522,17 @@ def ui_commit_url_anchor(self) -> str:
             return '<a href="#">n/a</a>'
         return f'<a href="{self.run.commit.commit_url}">{self.run.commit.hash[:7]}</a>'
 
+    @functools.cached_property
+    def unitsymbol(self) -> Optional[conbench.units.TUnit]:
+        """Return unit symbol or None if result indicates failure."""
+        if self.is_failed:
+            return None
+
+        # Not None, and not an empty string.
+        assert self.unit
+
+        return conbench.units.legacy_convert(self.unit)
```

Comment:
Let's try to use this in the future more than using `self.unit` directly, or make `self.unit` have a more precise type.